File: Sources/RehearsalLink/Models/RehearsalLinkProject.swift
import Foundation

struct RehearsalLinkProject: Codable {
    let audioFileURL: URL
    let segments: [AudioSegment]
    var summary: String? // プロジェクト全体の要約
    let createdAt: Date
    var modifiedAt: Date

    init(audioFileURL: URL, segments: [AudioSegment], summary: String? = nil) {
        self.audioFileURL = audioFileURL
        self.segments = segments
        self.summary = summary
        createdAt = Date()
        modifiedAt = Date()
    }
}

File: RehearsalLink-Mobile/src/models/RehearsalLinkProject.ts
import { AudioSegment } from './AudioSegment';

export interface RehearsalLinkProject {
  id: string; // Unique ID for the project folder
  name: string; // Display name
  audioFileName: string; // Name of the audio file in the project directory
  segments: AudioSegment[];
  summary?: string;
  fullTranscription?: string; // Added for full text
  createdAt: string; // ISO Date string
  modifiedAt: string; // ISO Date string
}
